src/main/scala/ch/openolitor/core/mailservice/MailService.scala

```diff
@@ -121,39 +129,77 @@ trait MailService extends AggregateRoot
     }
   }
 
-  def sendMail(meta: EventMetadata, uid: String, mail: Mail, commandMeta: Option[AnyRef]): Try[MailSentEvent] = {
+  def sendMail(meta: EventMetadata, uid: String, mail: Mail, commandMeta: Option[AnyRef]): Either[String, MailSentEvent] = {
+
+    val inputStreamfile = mail.attachmentReference.map { attachment: String =>
+      Await.result(fileStore.getFile(GeneriertBucket, attachment), 5 seconds)
+    }.getOrElse(Left(FileStoreError("Error")))
     if (sendEmailOutbound) {
-      var envelope = Envelope.from(new InternetAddress(fromAddress))
-        .to(InternetAddress.parse(mail.to): _*)
-        .subject(mail.subject)
-        .content(Text(mail.content))
+      var envelope = mail.attachmentReference match {
+        case Some(attachment) => {
+          inputStreamfile match {
+            case Right(f) => {
+              val file = new File("rechnung.pdf")
```

Comment:
Dangerous as this file might be overridden by concurrent execution of the same code! Add an id (rechnung) to the filename and you are save.

Reply:
Maybe use attachBytes and not even bother with a new File?

Reply:
good catch